client/client-multi/client-c/tools/kaa_encryption/rsa_key_gen/kaa_rsa_key_gen.c

```diff
@@ -153,13 +152,14 @@ int sha1_to_base64(uint8_t *key, size_t length, uint8_t *base64, size_t base64_l
 int kaa_keys_store(uint8_t *public_key, size_t public_key_length,
                           uint8_t *private_key, size_t private_key_length)
 {
-
+    
     FILE *fd = fopen(KAA_KEYS_STORAGE, "w");
     if (!fd) {
         return -1;
     }
 
     size_t written;
+    char buffer[512];
     fwrite(GUARD_IFNDEF, sizeof(GUARD_IFNDEF) - 1, 1, fd);
     fwrite(GUARD_DEF, sizeof(GUARD_DEF) - 1, 1, fd);
 
```

Comment:
Let's substitute `sprintf` with a combination of `snprintf` and assert that checks potential buffer overrun